llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -1112,6 +1237,68 @@ static unsigned TypeSizeToSizeIndex(TypeSize TS) {
   return Log2_32_Ceil((TypeSizeFixed + 7) / 8);
 }
 
+static bool isUnsupportedSPIRAccess(const Value *Addr, Instruction *I) {
+  // Skip SPIR-V built-in varibles
+  auto *OrigValue = Addr->stripInBoundsOffsets();
+  if (OrigValue->getName().starts_with("__spirv_BuiltIn"))
+    return true;
+
+  Type *PtrTy = cast<PointerType>(Addr->getType()->getScalarType());
+  switch (PtrTy->getPointerAddressSpace()) {
+  case kSpirOffloadPrivateAS:
+  case kSpirOffloadLocalAS:
+  case kSpirOffloadGenericAS:
+    return true;
+  }
+
+  return false;
+}
+
+static void noSanitizedSPIRInstruction(Instruction &I) {
```

Comment:
```suggestion
static void spirSetNoSanitizedMetadata(Instruction &I) {
```

I'm not 100% sure if my own suggestion is a good one, but I think that this function can be renamed into something which better represents what it does

Reply:
Makes sense. It's common to use verb as prefix of function name, so I changed it to "setNoSanitizedMetadataSPIR"